repair/repair.cc

```diff
@@ -1025,7 +1029,22 @@ static future<> repair_range(repair_info& ri, const dht::token_range& range) {
         return do_for_each(ri.cfs.begin(), ri.cfs.end(), [&ri, &neighbors, range] (auto&& cf) {
             ri._sub_ranges_nr++;
             if (ri.row_level_repair()) {
-                return repair_cf_range_row_level(ri, cf, range, neighbors);
+                if (ri.dropped_tables.count(cf)) {
+                    return make_ready_future<>();
+                }
+                return repair_cf_range_row_level(ri, cf, range, neighbors).handle_exception([&ri, cf] (std::exception_ptr ep) mutable {
+                    try {
+                        std::rethrow_exception(ep);
+                    } catch (const no_such_column_family& e) {
+                        if (!ri.ignore_dropped_tables) {
+                            ri.nr_failed_ranges++;
```

Comment:
The ignore_dropped_tables logic seems overly complicated.
Why not return exception future in this case?

Reply:
Because we want  the repair to continue to repair the result of the tables if one of the tables is dropped. If we return exception future here, the rest of the tables will be ignored.

Reply:
The logic is this:  if we see no_such_column_family exception, we always continue to repair the rest of the tables.  The ignore_dropped_tables controls if we treat the repair as a failure or success. 

For normal repair operations, ignore_dropped_tables = false, the repair will be treated an a failure. 

For repair based node operations, ignore_dropped_tables = true, the repair will be treated as a success.